### PR TITLE
Add C instr c.j

### PR DESCRIPTION
**Pull request by luyahan**

### CODE REVIEW

**Review by derekztu22 — APPROVED** [review]
lgtm

File: src/codegen/riscv64/assembler-riscv64-inl.h [hunk 1]
```diff
@@ -243,7 +243,10 @@ void Assembler::emit(ShortInstr x) {
   if (!is_buffer_growth_blocked()) {
     CheckBuffer();
   }
+  DEBUG_PRINTF("%p: ", pc_);
+  disassembleInstr(x);
   EmitHelper(x);
+  CheckTrampolinePoolQuick();
```
Comment (fw-brice) [inline src/codegen/riscv64/assembler-riscv64-inl.h:249]: Why does this emit call `CheckTrampolinePoolQuick` but the others do not?

File: src/codegen/riscv64/assembler-riscv64.h [hunk 2]
```diff
@@ -193,7 +193,8 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
     kOffset12 = 12,  // RISCV imm12
     kOffset20 = 20,  // RISCV imm20
     kOffset13 = 13,   // RISCV branch
-    kOffset32 = 32   // RISCV auipc + instr_I
+    kOffset32 = 32,   // RISCV auipc + instr_I
```
Comment (fw-brice) [inline src/codegen/riscv64/assembler-riscv64.h:196]: Please run `clang-format` on this file to fix the formatting:
```
clang-format -i -style=file src/codegen/riscv64/assembler-riscv64.h
```

File: src/diagnostics/riscv64/disasm-riscv64.cc [hunk 3]
```diff
@@ -1703,6 +1719,16 @@ void Decoder::DecodeCSType(Instruction* instr) {
   }
 }
 
+void Decoder::DecodeCJType(Instruction* instr) {
+  switch(instr->RvcOpcode()) {
+    case RO_C_J:
+      Format(instr, "jal       x0, 'Cimm11CJ");
```
Comment (fw-brice) [inline src/diagnostics/riscv64/disasm-riscv64.cc:1725]: `jal x0, offset` disassembles to `j offset`. Should this match?

File: test/cctest/test-disasm-riscv64.cc [hunk 4]
```diff
@@ -497,6 +497,8 @@ TEST(RV64C) {
   COMPARE(c_sw(a2, s1, 24), "0000cc90       sw       a2, 24(s1)");
   COMPARE(c_sd(a2, s1, 24), "0000ec90       sd       a2, 24(s1)");
   COMPARE(c_fsd(fa1, s1, 24), "0000ac8c       fsd       fa1, 24(s1)");
+
+  COMPARE(c_j(-12), "0000bfd5       jal       x0, -12");
```
Comment (fw-brice) [inline test/cctest/test-disasm-riscv64.cc:501]: See above comment about disassembly. Should this be `j -12`?

**Review by fw-brice — CHANGES_REQUESTED** [review]
Looks good. Just a few comments.

File: src/codegen/riscv64/assembler-riscv64-inl.h (hunk 1 above)
Comment (derekztu22) [reply src/codegen/riscv64/assembler-riscv64-inl.h:249]: @luyahan would you be able to answer this? I'm not too sure.

Comment (fw-brice) [reply src/codegen/riscv64/assembler-riscv64-inl.h:249]: @luyahan, can you answer this question? The other comments for this PR are resolved.

Comment (luyahan) [reply src/codegen/riscv64/assembler-riscv64-inl.h:249]: It is similiar with this:
```
void Assembler::emit(Instr x) {
  if (!is_buffer_growth_blocked()) {
    CheckBuffer();
  }
  DEBUG_PRINTF("%p: ", pc_);
  disassembleInstr(x);
  EmitHelper(x);
  CheckTrampolinePoolQuick();
}
```

Comment (fw-brice) [reply src/codegen/riscv64/assembler-riscv64-inl.h:249]: Sorry, you're right. I must've looked at this wrong before.

**Review by fw-brice — APPROVED** [review]

**Pull request merged**